assets/js/base/components/checkbox-control/index.js

```diff
@@ -55,9 +55,10 @@ const CheckboxControl = ( {
 				<path d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z" />
 			</svg>
 
-			<span className="wc-block-components-checkbox__label">
-				{ label }
-			</span>
+			<span
+				className="wc-block-components-checkbox__label"
+				dangerouslySetInnerHTML={ { __html: label } }
+			/>
```

Comment:
This so we can support dynamic labels that contain links, not sure if this is accessible.